src/System.Buffers/src/System/Buffers/DefaultArrayPoolBucket.cs

```diff
@@ -48,9 +54,19 @@ internal T[] Rent()
                 // Check if all of our buffers have been rented
                 if (_index < _data.Length)
                 {
-                    buffer = _data[_index] ?? new T[_bufferLength];
+                    buffer = _data[_index];
+                    if (buffer == null)
+                    {
+                        buffer = new T[_bufferLength];
+                        _eventSource.BufferCreated(buffer.GetHashCode(), _bufferLength, GetHashCode());
+                    }
                     _data[_index++] = null;
                 }
+                else if (_exhaustedEventSent == false)
+                {
+                    _eventSource.BucketExhausted(GetHashCode(), _bufferLength, _data.Length);
+                    _exhaustedEventSent = true;
+                }
```

Comment:
How is the data from BucketExhausted meant to be used differently from the data from BufferCreated?  It seems like we'd be logging a BufferCreated event in every case where BucketExhausted is logged?  If so, do we need BucketExhausted at all?